### PR TITLE
Add pragmas and improve observability

### PR DESCRIPTION
**Pull request by ptbrowne**
Following the update to stardog 8.1.0 (or is it 8.0.1) on int, we have to add a pragma `#pragma join.hash off` to our queries, specifically the observations one. To achieve this, I had to upgrade @tpluscode/sparql-builder and rdf-cube-view-query libraries for them to support the addition of pragmas to query.

Also added is a way to see query times on the client for better understanding of what is going on for each graphql query that we make: resolver fields are timed and shown as a flamegraph, and graphql queries that are made through our sparql client are timed.

The debugger for queries is available on develpment by default through an icon in the lower right corner. It is disabled by default on production, but can be made available through debug=true in the page URL.

- [ ] Some queries are not tracked, for example the one from rdf-cube-view-query.

### CODE REVIEW

**Review by bprusinowski — COMMENTED** [review]
The code looks nice, but I am not able to preview the branch locally (I have an `Error: Could not retrieve data: No dimensions` when previewing an Einmalvergütung für Photovoltaikanlagen dataset, and also data doesn't load for the charts). Did I miss something? 😱

File: app/rdf/create-source.ts [hunk 1]
```diff
@@ -1,10 +1,20 @@
+import { sparql } from "@tpluscode/sparql-builder";
 import { Source } from "rdf-cube-view-query";
 import rdf from "rdf-ext";
 
+const pragmas = `#pragma describe.strategy cbd
+#pragma join.hash off
+`;
+
+export const withPragma = (t: any) => {
+  return sparql`${pragmas}${t.toString()}`;
```
[the remark this answers is not in the document]
Comment (bprusinowski) [reply app/rdf/create-source.ts:10]: Or maybe it has to be like that, I missed the sparql prefix 😅

File: app/gql-flamegraph/devtool.tsx [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+              operation.query.definitions.find(
+                (d): d is OperationDefinitionNode =>
+                  d.kind === "OperationDefinition"
+              )?.name?.value
+            }
+            {result ? "✅" : "🔄"}
+          </Typography>
+          <Typography variant="caption">
+            {new Date(start).toISOString().slice(11, 19)}
+            {end ? `-${new Date(end).toISOString().slice(11, 19)}` : null}
```
[the remark this answers is not in the document]
Comment (ptbrowne) [reply app/gql-flamegraph/devtool.tsx]: I removed this as this was not super reliable.

File: app/rdf/create-source.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ptbrowne) [reply app/rdf/create-source.ts:10]: I removed this function as the new tpluscode/sparql-builder adds the prologue "method" that enables us to directly write

```
 SELECT`${fields}`.prologue`${pragmas}`
```

**Review by bprusinowski — APPROVED** [review]
Fantastic addition to introduce flame graph to see how the queries are made. Great job! 👏

**Pull request merged**